spec/models/sign_in/client_config_spec.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe SignIn::ClientConfig, type: :model do
+  let(:client_id) { SignIn::Constants::Auth::MOBILE_CLIENT }
+
+  describe 'validations' do
+    describe '#initialize' do
+      subject { SignIn::ClientConfig.new(client_id: client_id) }
+
+      context 'when client_id is nil' do
+        let(:client_id) { nil }
+        let(:expected_error_message) { 'Validation failed: Client can\'t be blank, Client is not included in the list' }
+        let(:expected_error) { ActiveModel::ValidationError }
+
+        it 'raises validation error' do
+          expect { subject }.to raise_error(expected_error, expected_error_message)
+        end
+      end
+
+      context 'when client_id is not in CLIENT_IDS constant' do
+        let(:client_id) { 'some-arbitrary-client-id' }
+        let(:expected_error_message) { 'Validation failed: Client is not included in the list' }
+        let(:expected_error) { ActiveModel::ValidationError }
+
+        it 'raises validation error' do
+          expect { subject }.to raise_error(expected_error, expected_error_message)
+        end
+      end
+    end
+  end
+
+  shared_examples 'client config methods' do |client_id, expected_values|
+    let(:client_config) { SignIn::ClientConfig.new(client_id: client_id) }
+
+    describe '#cookie_auth?' do
+      it 'matches expected cookie auth value' do
+        expect(client_config.cookie_auth?).to eq(expected_values[:cookie_auth])
+      end
+    end
+
+    describe '#api_auth?' do
+      it 'matches expected api auth value' do
+        expect(client_config.api_auth?).to eq(expected_values[:api_auth])
+      end
+    end
+
+    describe '#anti_csrf?' do
+      it 'matches expected anti csrf value' do
+        expect(client_config.anti_csrf?).to eq(expected_values[:anti_csrf])
+      end
+    end
+
+    describe '#redirect_uri' do
+      it 'matches expected redirect uri value' do
+        expect(client_config.redirect_uri).to eq(expected_values[:redirect_uri])
+      end
+    end
+
+    describe '#access_token_duration' do
+      it 'matches expected access token duration value' do
+        expect(client_config.access_token_duration).to eq(expected_values[:access_token_duration])
+      end
+    end
+
+    describe '#access_token_audience' do
+      it 'matches expected access token audience value' do
+        expect(client_config.access_token_audience).to eq(expected_values[:access_token_audience])
+      end
+    end
+
+    describe '#refresh_token_duration' do
+      it 'matches expected refresh token duration value' do
+        expect(client_config.refresh_token_duration).to eq(expected_values[:refresh_token_duration])
+      end
+    end
+  end
+
+  describe 'client config methods' do
+    context 'mobile auth' do
+      it_behaves_like 'client config methods',
```

Comment:
Instead of passing arguments into the `shared examples`, a more rspec way would be to define `let` values, like this:

> context 'mobile auth' do
>   let(:client_id) { SignIn::Constants::Auth::MOBILE_CLIENT }
>   let(:expected_values) { ... }
>
>   it_behaves_like 'client config methods'